grudge/discretization.py

```diff
@@ -21,45 +30,55 @@
 """
 
 from pytools import memoize_method
+
 from grudge.dof_desc import (
     DISCR_TAG_BASE, DISCR_TAG_MODAL,
     DTAG_BOUNDARY, DOFDesc, as_dofdesc
 )
+
 import numpy as np  # noqa: F401
-from meshmode.array_context import ArrayContext
-from meshmode.discretization.connection import \
-    FACE_RESTR_INTERIOR, FACE_RESTR_ALL, make_face_restriction
-from meshmode.mesh import BTAG_PARTITION
 
-from warnings import warn
+from arraycontext.context import ArrayContext
 
+from meshmode.discretization.connection import (
+    FACE_RESTR_INTERIOR,
+    FACE_RESTR_ALL,
+    make_face_restriction
+)
+from meshmode.mesh import BTAG_PARTITION
 
-__doc__ = """
-.. autoclass:: DiscretizationCollection
-"""
+from warnings import warn
 
 
 class DiscretizationCollection:
-    """
-    .. automethod :: __init__
+    """A collection of discretizations on various mesh entities
+    (volume, interior facets, boundaries) and associated element
+    groups.
+
+    .. automethod:: __init__
 
-    .. automethod :: discr_from_dd
-    .. automethod :: connection_from_dds
+    .. autoattribute:: dim
+    .. autoattribute:: ambient_dim
+    .. autoattribute:: mesh
+    .. autoattribute:: real_dtype
+    .. autoattribute:: complex_dtype
 
-    .. autoattribute :: dim
-    .. autoattribute :: ambient_dim
-    .. autoattribute :: mesh
+    .. automethod:: discr_from_dd
+    .. automethod:: connection_from_dds
 
-    .. automethod :: empty
-    .. automethod :: zeros
+    .. automethod:: empty
+    .. automethod:: zeros
     """
 
     def __init__(self, array_context, mesh, order=None,
             discr_tag_to_group_factory=None, mpi_communicator=None,
             # FIXME: `quad_tag_to_group_factory` is deprecated
             quad_tag_to_group_factory=None):
-        """
-        :param discr_tag_to_group_factory: A mapping from discretization tags
+        """Constructor for the :class:`DiscretizationCollection` object.
+
+        :arg actx: an :class:`~arraycontext.context.ArrayContext`.
+        :arg mesh: a :class:`~meshmode.mesh.Mesh` object.
```

Comment:
Add some types instead of docs?

Reply:
Done: https://github.com/inducer/grudge/commit/52c7318c777efebf2c559986f0f7b6a429898e89